### PR TITLE
Migrate Java unit tests to JUnit5

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

See the title

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- Unit tests were migrated

- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/17452

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by renner — APPROVED** [review]
Great patch, thanks a lot! Only nitpicky suggestions about additional empty lines that likely were created by automatic conversion, or maybe even on purpose sometimes, and I might have overlooked some as well. Next we might be ready to work on the directory structure, i.e. move all test to a separate directory?

File: java/buildconf/test/rhn.conf.postgresql-example [hunk 1]
```diff
@@ -37,8 +37,8 @@ web.mount_point = /pub/
 # Mail settings
 web.disallowed_mail_domains =
 web.restrict_mail_domains =
-traceback_mail = galaxy-noise@suse.de
-web.traceback_mail = galaxy-noise@suse.de
+traceback_mail = fkobzik-local
+web.traceback_mail = fkobzik-local
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply java/buildconf/test/rhn.conf.postgresql-example:41]: nice catch

File: java/code/src/com/redhat/rhn/common/client/test/ClientCertificateDigesterTest.java [hunk 2]
```diff
@@ -14,37 +14,47 @@
  */
 package com.redhat.rhn.common.client.test;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/common/client/test/ClientCertificateDigesterTest.java:21]: 👍🏻

File: java/code/src/com/redhat/rhn/common/db/datasource/test/UserData.java [hunk 3]
```diff
@@ -14,6 +14,7 @@
  */
 package com.redhat.rhn.common.db.datasource.test;
 
+
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/common/db/datasource/test/UserData.java:17]: 👍🏻

File: java/code/src/com/redhat/rhn/common/finder/test/FileFinderTest.java [hunk 4]
```diff
@@ -14,19 +14,29 @@
  */
 
 package com.redhat.rhn.common.finder.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/common/finder/test/FileFinderTest.java:22]: 👍🏻

**Pull request merged**